coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java

```diff
@@ -255,6 +262,24 @@ public RexNode visitCall(RexCall call) {
       return adjustReturnTypeWithCast(rexBuilder, super.visitCall(modifiedCall));
     }
 
+    private Optional<RexNode> visitConcat(RexCall call) {
+      // Hive supports operations like CONCAT(date, varchar) while Trino only supports CONCAT(varchar, varchar)
+      // So we need to cast the unsupported types to varchar
+      final SqlOperator op = call.getOperator();
+      List<RexNode> convertedOperands = visitList(call.getOperands(), (boolean[]) null);
+      List<RexNode> castOperands = new ArrayList<>();
+
+      for (RexNode inputOperand : convertedOperands) {
+        if (inputOperand.getType().getSqlTypeName() != VARCHAR && inputOperand.getType().getSqlTypeName() != CHAR) {
```

Comment:
will this condition always be false? can getType().getSqlTypeName() be both VARCHAR and CHAR?

can this be changed to - 
!Sets.newHashSet(SqlTypeName.VARCHAR, SqlTypeName.CHAR).contains(inputOperand.getType().getSqlTypeName())

Reply:
okay, yeah sorry my bad.
technically this implementation is also correct.